integrationTests/chainSimulator/staking/simpleStake_test.go

```diff
@@ -0,0 +1,133 @@
+package staking
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	"github.com/multiversx/mx-chain-go/vm"
+	"github.com/stretchr/testify/require"
+)
+
+// Test scenarios
+// Do 3 stake transactions from 3 different wallets - tx value 2499, 2500, 2501
+// testcase1 -- staking v3.5 --> tx1 fail, tx2 - node in queue, tx3 - node in queue with topUp 1
+// testcase2 -- staking v4 step1 --> tx1 fail, tx2 - node in auction, tx3 - node in auction with topUp 1
+// testcase3 -- staking v4 step2 --> tx1 fail, tx2 - node in auction, tx3 - node in auction with topUp 1
+// testcase4 -- staking v3.step3 --> tx1 fail, tx2 - node in auction, tx3 - node in auction with topUp 1
+func TestChainSimulator_SimpleStake(t *testing.T) {
+	t.Run("staking ph 4 is not active", func(t *testing.T) {
+		testChainSimulatorSimpleStake(t, 1, "queued")
+	})
+
+	t.Run("staking ph 4 step1", func(t *testing.T) {
+		testChainSimulatorSimpleStake(t, 2, "auction")
+	})
+
+	t.Run("staking ph 4 step2", func(t *testing.T) {
+		testChainSimulatorSimpleStake(t, 3, "auction")
+	})
+
+	t.Run("staking ph 4 step3", func(t *testing.T) {
+		testChainSimulatorSimpleStake(t, 4, "auction")
+	})
+}
+
+func testChainSimulatorSimpleStake(t *testing.T, targetEpoch int32, nodesStatus string) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+	startTime := time.Now().Unix()
+	roundDurationInMillis := uint64(6000)
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	numOfShards := uint32(3)
+
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   false,
+		TempDir:                  t.TempDir(),
+		PathToInitialConfig:      defaultPathToInitialConfig,
+		NumOfShards:              numOfShards,
+		GenesisTimestamp:         startTime,
+		RoundDurationInMillis:    roundDurationInMillis,
+		RoundsPerEpoch:           roundsPerEpoch,
+		ApiInterface:             api.NewNoApiInterface(),
+		MinNodesPerShard:         3,
+		MetaChainMinNodes:        3,
+		NumNodesWaitingListMeta:  3,
+		NumNodesWaitingListShard: 3,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.EpochConfig.EnableEpochs.StakingV4Step1EnableEpoch = 2
```

Comment:
we might create a "helper" function for these 4 sets by having a "starting" epoch as parameter

Reply:
done